packages/hoppscotch-common/src/newstore/settings.ts

```diff
@@ -190,6 +204,19 @@ export function toggleSetting(settingKey: KeysMatching<SettingsDef, boolean>) {
   })
 }
 
+export function toggleNestedSetting<
+  K extends KeysMatching<SettingsDef, object>,
+  P extends keyof SettingsDef[K],
+>(settingKey: K, property: P) {
+  settingsStore.dispatch({
+    dispatcher: "toggleNestedSetting",
+    payload: {
+      settingKey,
+      property,
```

Comment:
There's a type error arising here.

Reply:
> There's a type error arising here.

Added // @ts-error here as TypeScript is not able to figure out the passed property